File: 2021/01/31/jh20s/contest#225/1738.cpp
class Solution {
public:
int dp[1005][1005];
int kthLargestValue(vector<vector<int>>& matrix, int k) {
	dp[0][0] = matrix[0][0];
	vector<int> v;
	for (int i = 0; i < matrix[0].size(); i++) {
		dp[0][i] = matrix[0][i];
	}
	for (int i = 1; i < matrix.size(); i++) {
		for(int j=0;j<matrix[i].size();j++){
			dp[i][j] = dp[i-1][j] ^ matrix[i][j];
		}
	}
	for (int i = 0; i < matrix.size(); i++) {
		for (int j = 1; j < matrix[i].size(); j++) {
			dp[i][j] = dp[i][j] ^ dp[i][j - 1];
		}
	}
	for (int i = 0; i < matrix.size(); i++) {
		for (int j = 0; j < matrix[i].size(); j++) {
			v.push_back(dp[i][j]);
		}
	}
	sort(v.begin(), v.end(),greater<int>());
	return v[k-1];
}
};

File: 2021/01/31/jh20s/contest#225/1738.py
class Solution:
    def kthLargestValue(self, matrix: List[List[int]], k: int) -> int:
        m, n = len(matrix), len(matrix[0]) # dimensions 
        
        ans = []
        for i in range(m): 
            for j in range(n): 
                if i: matrix[i][j] ^= matrix[i-1][j]
                if j: matrix[i][j] ^= matrix[i][j-1]
                if i and j: matrix[i][j] ^= matrix[i-1][j-1]
                ans.append(matrix[i][j])
        return sorted(ans)[-k] 